text/0000-light-dom.md

```diff
@@ -0,0 +1,218 @@
+
+As we mentioned before, the component composition model in LWC is provided by slots. Light DOM will provide the same mental model for developers building Light DOM components.
+
+In Light DOM, `<slot>` will denote the place where the slotted component will be attached. The `<slot>` element itself won't be rendered. The slotted content (or the fallback content) will be flattened to the parent element at runtime.
+
+Since the `<slot>` element itself isn't rendered, adding attributes or event listeners to the `<slot>` element in the template will throw a compiler error.
+
+#### Styles
+
+Until now styles used in LWC components where scoped to the component thanks to shadow DOM (or synthetic shadow DOM) style scoping. In the light DOM, component styles are naturally leaks out of the component, LWC doesn't do any style scoping out of the box. Developers are in charge of making sure to write specific enough selector to target the component content.
+
+To support the cases where a shadow DOM element composes a light element, light DOM styles are required to be injected to the closest root node. For a given light DOM elements if all the ancestor components are also a light DOM components, the component style sheet will be injected in the document `<head>`. Otherwise if any of the ancestors is a shadow DOM components the style has to be injected in the closest shadow root. Upon insertion of a light DOM element does the following steps:
+
+- look up for the closest root node (`Node.prototype.getRootNode()`)
+- insert the stylesheet if not already present:
+  - if the root node is the HTML document, the style sheet is inserted as a `<style>` element in the `<head>`.
+  - if the root node is a shadow root, the stylesheet is inserted as a `<style>` element as the first shadow root child.
+
+It is important to notice that the order in which light DOM components are rendered impact the order in which stylesheets are injected in the root node and directly influences CSS rule specificity.
+
+**Note:** Different approaches layer style scoping on top has been discussed while designing Light DOM, like introducing a new file extension for automatic style scoping `.scoped.css` or using a `<style scoped>` element in the template.
+
+#### `this.template`
+
+In `LightningElement`, `this.template` returns the shadow-root. It will return `null` in `MacroElement`.
+
+### Querying
+
+`MacroElement` (and `LightningElement`) forward several DOM querying/manipulation methods like `querySelector`, `dispatchEvent` etc. to the host element. Full list of methods forwarded [here](https://github.com/salesforce/lwc/blob/master/packages/@lwc/engine-core/src/framework/base-lightning-element.ts#L242).
+
+This will allow component author to just use `this.querySelector` instead of using `this.template.querySelector` in case of `LightningElement`.
+However, there's no way to get a direct reference to the host element, like `this.template.host` that's available in `LightningElement`.
```

Comment:
I'm confused by this section. In the context of a light DOM component, `this` refers to the host element, correct? (At least, it does with vanilla web components.)

I'm also not sure what the example below is supposed to be illustrating? Is the example below about event retargeting when a light DOM component has a shadow DOM component as its child? What is the example communicating? (Maybe some code comments would help. 🙂 )